neo/Consensus/ConsensusContext.cs

```diff
@@ -24,19 +24,22 @@ internal class ConsensusContext : IConsensusContext
         public ECPoint[] Validators { get; set; }
         public int MyIndex { get; set; }
         public uint PrimaryIndex { get; set; }
+        public ConsensusPayload[] ChangeViewPayloads { get; set; }
         public uint Timestamp { get; set; }
         public ulong Nonce { get; set; }
         public UInt160 NextConsensus { get; set; }
         public UInt256[] TransactionHashes { get; set; }
         public Dictionary<UInt256, Transaction> Transactions { get; set; }
         public UInt256[] Preparations { get; set; }
+        public ConsensusPayload[] PreparationPayloads { get; set; }
         public byte[][] Commits { get; set; }
         public byte[] ExpectedView { get; set; }
```

Comment:
Since we have `ChangeViewPayloads`, we can remove `ExpectedView`.

Reply:
I'm in the middle of removing ExpectedView; it is mostly straight forward, but there is one part a bit tricky, because in recovery it doesn't keep the `ChangeView` messages if it has enough `PreparationPayloads`. It is possible it could have `M` `PreparationPayloads` but not the `PrepareRequest`; in that case it wants to know each of the `ExpectedView` values for all the nodes, but it won't have their `ChangeViewPayloads` in that case. I think the code was missing setting the `ExpectedView` values in that case already. What happens in that case is it would potentially save an older ChangeView message. Maybe it is better to the `ExpectedView` so we can keep the optimization of not sending them in the recovery message if there are >= `M` preparations. I'm looking into solutions now.

Reply:
> because in recovery it doesn't keep the `ChangeView` messages if it has enough `PreparationPayloads`.

I think in `RecoveryMessage`, we must first check the `ChangeView` messages for view changing. After that, then we can check other messages such as `PrepareRequest` and `PrepareResponse`.

Reply:
I think we can simply think of `RecoveryMessage` as a kind of replay of `ChangeView`, `PrepareRequest`, `PrepareResponse`, and `Commit` messages. This will make the logic very clear and not prone to errors.

Reply:
Done.